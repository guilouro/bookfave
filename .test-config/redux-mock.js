import React from 'react'
import { createStore, combineReducers, applyMiddleware } from 'redux'
import { render } from 'react-testing-library'
import { Provider } from 'react-redux'

export const mockStore = mock => {
  const reducers = {}

  Object.keys(mock).forEach(key => {
    reducers[key] = () => mock[key]
  })

  return createStore(combineReducers(reducers))
}

export const renderWithRedux = (component, store) =>
  render(<Provider store={store}>{component}</Provider>)
