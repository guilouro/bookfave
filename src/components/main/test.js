import React from 'react'
import Main from '.'
import { mockStore, renderWithRedux } from '../../../.test-config/redux-mock'

const store = mockStore({
  favorites: {
    filterTag: '',
    entities: {},
    ids: []
  }
})

describe('Tag', () => {
  it('Should render correctly', () => {
    const { container } = renderWithRedux(<Main />, store)
    expect(container).toMatchSnapshot()
  })
})
