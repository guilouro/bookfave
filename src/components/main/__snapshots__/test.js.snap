// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Should render correctly 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  padding: 20px 0;
  text-align: center;
}

.c2 {
  color: #b0b9bf;
  padding: 10px;
}

.c3 {
  padding: 20px;
  background: #fff;
  border-radius: 10px;
}

.c10 {
  margin-top: 20px;
  height: 100%;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  width: 70px;
  height: 35px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 10px;
}

.c6 {
  outline: none;
  border: none;
  background: transparent;
  margin: 0;
  padding: 0;
  cursor: pointer;
  position: absolute;
  color: #b0b9bf;
  -webkit-transition: 0.5s all;
  transition: 0.5s all;
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

.c7 {
  outline: none;
  border: none;
  background: transparent;
  margin: 0;
  padding: 0;
  cursor: pointer;
  position: absolute;
  color: #1e68c1;
  -webkit-transition: 0.5s all;
  transition: 0.5s all;
  -webkit-transform: translateX(80%);
  -ms-transform: translateX(80%);
  transform: translateX(80%);
}

.c8 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c8 button {
  background-color: #1e68c1;
  color: #fff;
  border: none;
  padding: 10px;
  border-radius: 5px;
}

.c8 button:disabled {
  background-color: #b0b9bf;
}

.c9 {
  border: 1px solid #e7e8eb;
  border-radius: 4px;
  padding: 8px 10px;
  margin-bottom: 10px;
}

.c9::-webkit-input-placeholder {
  color: #b0b9bf;
}

.c9::-moz-placeholder {
  color: #b0b9bf;
}

.c9:-ms-input-placeholder {
  color: #b0b9bf;
}

.c9::placeholder {
  color: #b0b9bf;
}

.c11 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #b0b9bf;
  padding-top: 50px;
}

@media (min-width:900px) {
  .c0 {
    width: 900px;
  }
}

@media (min-width:900px) {
  .c1 {
    text-align: left;
  }
}

@media (min-width:900px) {
  .c10 {
    height: 400px;
    overflow-y: overlay;
  }
}

@media (min-width:900px) {
  .c4 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:600px) {
  .c8 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:900px) {
  .c8 button {
    padding-top: 0;
    padding-bottom: 0;
  }
}

@media (min-width:600px) {
  .c9 {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
    margin: 0;
  }

  .c9:not(:last-child) {
    margin-right: 15px;
  }
}

<main
  class="c0"
>
  <header
    class="c1"
  >
    <img
      alt="BOOKFAVE"
      src="test-file-stub"
    />
    <span
      class="c2"
    >
      Challenge
    </span>
  </header>
  <section
    class="c3"
  >
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <button
          class="c6"
        >
          <svg
            height="20px"
            viewBox="0 0 26 26"
            width="20px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.583 20.083c-4.687 0-8.5-3.812-8.5-8.5 0-4.687 3.813-8.5 8.5-8.5 4.688 0 8.5 3.813 8.5 8.5 0 4.688-3.812 8.5-8.5 8.5m13.752 3.249l-4.81-4.811a11.265 11.265 0 0 0 2.391-6.938C22.916 5.334 17.832.25 11.583.25 5.334.25.25 5.334.25 11.583c0 6.25 5.084 11.334 11.333 11.334 2.615 0 5.02-.9 6.937-2.392l4.811 4.81a1.413 1.413 0 0 0 2.004 0 1.415 1.415 0 0 0 0-2.003"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          class="c7"
        >
          <svg
            height="20px"
            viewBox="0 0 26 26"
            width="20px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.121 11.72h-3.84V7.879a1.28 1.28 0 0 0-2.561 0v3.84H7.879a1.28 1.28 0 0 0 0 2.561h3.84v3.841a1.28 1.28 0 0 0 2.561 0v-3.84h3.841a1.28 1.28 0 0 0 0-2.561zm5.121 7.682a3.845 3.845 0 0 1-3.84 3.84H6.598a3.845 3.845 0 0 1-3.84-3.84V6.598a3.845 3.845 0 0 1 3.84-3.84h12.804a3.845 3.845 0 0 1 3.84 3.84v12.804zM19.402.197H6.598A6.409 6.409 0 0 0 .197 6.598v12.804a6.409 6.409 0 0 0 6.401 6.401h12.804a6.409 6.409 0 0 0 6.401-6.401V6.598A6.409 6.409 0 0 0 19.402.197z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <form
        novalidate=""
      >
        <div
          class="c8"
        >
          <input
            class="c9"
            name="title"
            placeholder="Title"
            required=""
            value=""
          />
          <input
            class="c9"
            name="link"
            placeholder="Link"
            required=""
            value=""
          />
          <input
            class="c9"
            name="tags"
            placeholder="Tags"
            value=""
          />
          <button
            type="submit"
          >
            Add
          </button>
        </div>
      </form>
    </div>
    <div
      class="c10"
    >
      <div
        class="c11"
      >
        <svg
          fill="currentColor"
          height="50px"
          viewBox="0 0 475 475.884"
          width="50px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M473.738 328.543A23.656 23.656 0 0 0 450.922 312h-10.48V56a8 8 0 0 0-8-8h-112V8a8 8 0 0 0-8-8h-304a8 8 0 0 0-8 8v416a8 8 0 0 0 8 8h264.32l-4 12.336a23.993 23.993 0 0 0 8.695 26.879 24 24 0 0 0 28.25-.016l41.934-30.543a8.06 8.06 0 0 1 9.511 0l42.012 30.559a23.655 23.655 0 0 0 28.223 0 23.66 23.66 0 0 0 8.718-26.871l-16.062-49.489a7.915 7.915 0 0 1 2.934-8.863l42.09-30.558a23.71 23.71 0 0 0 8.671-26.891zM424.441 64v248h-25.52a7.941 7.941 0 0 1-2.472-.406 8.15 8.15 0 0 1-5.117-5.106L375.25 257.04a24.002 24.002 0 0 0-45.602 0l-9.207 28.266V208h64a8 8 0 0 0 0-16h-64v-32h64a8 8 0 0 0 0-16h-64v-32h64a8 8 0 0 0 0-16h-64V64zm-408 352V16h288v296h-50.48a24.004 24.004 0 0 0-22.879 16.566 24.01 24.01 0 0 0 8.734 26.868l42.075 30.55a7.927 7.927 0 0 1 2.945 8.903L277.969 416zm439.2-73.504l-42.079 30.559a23.87 23.87 0 0 0-8.738 26.777l16.082 49.441a7.999 7.999 0 0 1-12.32 8.973l-42.04-30.582a24.2 24.2 0 0 0-28.265 0l-41.992 30.559a7.999 7.999 0 1 1-12.328-8.965l16.074-49.418a23.883 23.883 0 0 0-8.746-26.817l-42.047-30.52A8.002 8.002 0 0 1 253.961 328h52a22.367 22.367 0 0 0 9.648-2.098l.153-.086.144-.046c.114-.051.192-.145.305-.204a23.754 23.754 0 0 0 12.559-14.132L344.852 262a8.003 8.003 0 0 1 7.597-5.504 8.004 8.004 0 0 1 7.602 5.504l16.086 49.473A24.006 24.006 0 0 0 398.922 328h52a7.995 7.995 0 0 1 7.664 5.52 7.994 7.994 0 0 1-2.938 8.976zm0 0"
          />
          <path
            d="M72.441 80h176a8 8 0 0 0 0-16h-176a8 8 0 0 0 0 16zm0 0M72.441 144h176a8 8 0 0 0 0-16h-176a8 8 0 0 0 0 16zm0 0M72.441 208h176a8 8 0 0 0 0-16h-176a8 8 0 0 0 0 16zm0 0M256.441 264a8 8 0 0 0-8-8h-176a8 8 0 0 0 0 16h176a8 8 0 0 0 8-8zm0 0"
          />
        </svg>
        <p>
          We don't have any favorite
        </p>
      </div>
    </div>
  </section>
</main>
`;
